rust/automerge/src/automerge.rs

```diff
@@ -1432,6 +1432,34 @@ impl ReadDoc for Automerge {
             .get(hash)
             .and_then(|index| self.history.get(*index))
     }
+
+    fn hash_for_opid(&self, opid: &ExId) -> Option<ChangeHash> {
+        match opid {
```

Comment:
You need to convert the ExId to an ObjId via `let obj = self.exid_to_obj(obj)?;`  The reason being is that if the ExId was created by parsing a string the actor_id will be correct but the actor_index will not be.  The `exid_to_obj()` method sorts this out.

Reply:
I've added a `exid_to_opid`, which is what that `exid_to_obj` did internally that gets us what we need for this function. Otherwise we get errors for querying the hashes of non-object ops